src/Native/System.Native/pal_networking.h

```diff
@@ -409,7 +409,7 @@ extern "C" Error CreateSocketEventBuffer(int32_t count, SocketEvent** buffer);
 
 extern "C" Error FreeSocketEventBuffer(SocketEvent* buffer);
 
-extern "C" Error TryChangeSocketEventRegistration(int32_t port, int32_t socket, int32_t currentEvents, int32_t newEvents, uintptr_t data);
+extern "C" Error TryChangeSocketEventRegistration(int32_t port, int32_t socket, int32_t currentEvents, int32_t newEvents, int64_t data);
```

Comment:
The size of this parameter shouldn't be changed in any case: these `data` params are intentionally pointer-sized (the values are derived from `GCHandle`s).